### ISSUE TITLE
FlushError: Can't update table using NULL for primary key value

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1371]
_Imported from trac ticket 1371.
Created by **kbower**
Opened in trac: Jan 16, 2012, 2:31:33 PM
Last modified in trac: Jan 19, 2012, 8:35:52 AM_
***
For orders with comments written in gers

See http://www.sqlalchemy.org/trac/ticket/2127#comment:9

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 19, 2012, 8:34:56 AM:_

(Changeset [[2186]](../commit/33f6a0e581b934295d060753d12f12b18d0df0f8))
 * Fix #1371 (FlushError: Can't update table using NULL for primary key value)

```diff
diff -rU5 SQLAlchemy-0.7.4.92kb/lib/sqlalchemy/__init__.py SQLAlchemy-0.7.4.93kb/lib/sqlalchemy/__init__.py
--- SQLAlchemy-0.7.4.92kb/lib/sqlalchemy/__init__.py    2012-01-12 12:11:28.000000000 -0500
+++ SQLAlchemy-0.7.4.93kb/lib/sqlalchemy/__init__.py    2012-01-18 16:04:12.000000000 -0500
@@ -115,11 +115,11 @@


 __all__ = sorted(name for name, obj in locals().items()
                  if not (name.startswith('_') or inspect.ismodule(obj)))

-__version__ = '0.7.4.92kb'
+__version__ = '0.7.4.93kb'

 del inspect, sys

 from sqlalchemy import util as _sa_util
 _sa_util.importlater.resolve_all()
diff -rU5 SQLAlchemy-0.7.4.92kb/lib/sqlalchemy/orm/mapper.py SQLAlchemy-0.7.4.93kb/lib/sqlalchemy/orm/mapper.py
--- SQLAlchemy-0.7.4.92kb/lib/sqlalchemy/orm/mapper.py  2011-12-15 11:42:50.000000000 -0500
+++ SQLAlchemy-0.7.4.93kb/lib/sqlalchemy/orm/mapper.py  2012-01-18 16:06:25.000000000 -0500
@@ -2177,11 +2177,11 @@
                                 else:
                                     hasdata = True
                             elif col in pks:
                                 value = state.manager[prop.key].\
                                             impl.get(state, state_dict)
-                                if value is None:
+                                if value is None and not mapper.allow_partial_pks:
                                     raise sa_exc.FlushError(
                                                 "Can't update table "
                                                 "using NULL for primary "
                                                 "key value")
                                 params[col._label] = value
```

**Comment by kbower** [comment]
_**kbower** trac status change on Jan 19, 2012, 8:34:56 AM:_ **closed**

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 19, 2012, 8:35:52 AM:_

NOTE: Not fixed at this time in sqlalchemy, see [#1382](1382)
